src/main/java/com/yolo/customer/userProfile/UserProfileController.java

```diff
@@ -20,4 +20,15 @@ public ResponseEntity<String> createUserProfile(
         UserProfile userProfile = userProfileService.createUserProfile(username, userProfileRequest);
         return ResponseEntity.ok("User profile created successfully.");
     }
+
+
+    @PreAuthorize("hasAuthority('ROLE_UPDATE_PROFILE')")
+    @PatchMapping("/{username}/profiles")
+    public ResponseEntity<String> updateUserProfile(
```

Comment:
use    Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        String username = GetContextHolder.getUsernameFromAuthentication(authentication);
        User loggedInUser = userRepository.findByUsername(username).get();

        if(loggedInUser == null) {
            throw new IllegalArgumentException("User with given username doesnot exists: " + username);
        } instead of exposing username in URL